database/vertica/vertica.go

```diff
@@ -0,0 +1,336 @@
+package vertica
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"io"
+	"io/ioutil"
+	nurl "net/url"
+
+	"github.com/golang-migrate/migrate/v4"
+	"github.com/golang-migrate/migrate/v4/database"
+	"github.com/hashicorp/go-multierror"
+	_ "github.com/vertica/vertica-sql-go"
+)
+
+func init() {
+	database.Register("vertica", &Vertica{})
+}
+
+var DefaultMigrationsTable = "schema_migrations"
+
+var (
+	ErrDatabaseDirty    = fmt.Errorf("database is dirty")
+	ErrNilConfig        = fmt.Errorf("no config")
+	ErrNoDatabaseName   = fmt.Errorf("no database name")
+	ErrAppendPEM        = fmt.Errorf("failed to append PEM")
+	ErrTLSCertKeyConfig = fmt.Errorf("To use TLS client authentication, both x-tls-cert and x-tls-key must not be empty")
+)
+
+type Config struct {
+	MigrationsTable string
+	DatabaseName    string
+	Schema          string
+}
+
+type Vertica struct {
+	conn   *sql.Conn
+	db     *sql.DB
+	config *Config
+}
+
+func (v *Vertica) Open(url string) (database.Driver, error) {
+	purl, err := nurl.Parse(url)
+	db, err := sql.Open("vertica", migrate.FilterCustomQuery(purl).String())
+	if err != nil {
+		return nil, err
+	}
+
+	migrationsTable := purl.Query().Get("x-migrations-table")
+	if len(migrationsTable) == 0 {
+		migrationsTable = DefaultMigrationsTable
+	}
+	schema := purl.Query().Get("x-schema")
+
+	vx, err := WithInstance(db, &Config{
+		Schema:          schema,
+		DatabaseName:    parseDbName(url),
+		MigrationsTable: migrationsTable,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return vx, nil
+}
+
+func (v *Vertica) Close() error {
+	connErr := v.conn.Close()
+	dbErr := v.db.Close()
+	if connErr != nil || dbErr != nil {
+		return fmt.Errorf("conn: %v, db: %v", connErr, dbErr)
+	}
+	return nil
+}
+
+func (v *Vertica) dropTables() error {
+	// select all tables
+	whereStmt := ""
+	if len(v.config.Schema) > 0 {
+		whereStmt = ` WHERE table_schema='` + v.config.Schema + `'`
+	}
+	query := `SELECT DISTINCT table_name FROM tables` + whereStmt
```

Comment:
I'm not familiar with Vertica. Is this using the [v_catalog](https://www.vertica.com/docs/9.2.x/HTML/Content/Authoring/SQLReferenceManual/SystemTables/CATALOG/V_CATALOGSchema.htm?tocpath=SQL%20Reference%20Manual%7CVertica%20System%20Tables%7CV_CATALOG%20Schema%7C_____0) schema?

Is there another [DROP statement](https://www.vertica.com/docs/9.2.x/HTML/Content/Authoring/SQLReferenceManual/Statements/DropStatements.htm?tocpath=SQL%20Reference%20Manual%7CSQL%20Statements%7CDROP%20Statements%7C_____0) we could use instead? e.g. `DROP SCHEMA`

Reply:
No this is not v_catalog.  In the projects  I have worked with, schemas house tables views, etc.  My projects are typically scoped to schema.  What this code  is doing is scoping the migrations to the schema. 

Think of vertica schema's like mariadb databases.

Reply:
If this isn't [v_catalog](https://www.vertica.com/docs/9.2.x/HTML/Content/Authoring/SQLReferenceManual/SystemTables/CATALOG/V_CATALOGSchema.htm?tocpath=SQL%20Reference%20Manual%7CVertica%20System%20Tables%7CV_CATALOG%20Schema%7C_____0) then where is the `tables` table being managed?

Reply:
I apologize, I thought you were referencing v.config object being v_catalog. 

tables, all_tables  comes from v_catalog